include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_fpga.h

```diff
@@ -77,218 +77,36 @@ struct __parallel_for_fpga_submitter<__internal::__optional_kernel_name<_Name...
-    -> oneapi::dpl::__internal::__enable_if_fpga_execution_policy<
-        _ExecutionPolicy, decltype(oneapi::dpl::__par_backend_hetero::__parallel_merge(
-                              __exec.__device_policy(), ::std::forward<_Range1>(__rng1),
-                              ::std::forward<_Range2>(__rng2), ::std::forward<_Range3>(__rng3), __comp))>
-{
-    // workaround until we implement more performant version for patterns
-    return oneapi::dpl::__par_backend_hetero::__parallel_merge(
-        __exec.__device_policy(), ::std::forward<_Range1>(__rng1), ::std::forward<_Range2>(__rng2),
-        ::std::forward<_Range3>(__rng3), __comp);
-}
-
-//------------------------------------------------------------------------
-// parallel_stable_sort
-//-----------------------------------------------------------------------
-
-template <typename _ExecutionPolicy, typename _Range, typename _Compare, typename _Proj,
-          oneapi::dpl::__internal::__enable_if_fpga_execution_policy<_ExecutionPolicy, int> = 0>
-auto
-__parallel_stable_sort(_ExecutionPolicy&& __exec, _Range&& __rng, _Compare __comp, _Proj __proj)
-{
-    // workaround until we implement more performant version for patterns
-    return oneapi::dpl::__par_backend_hetero::__parallel_stable_sort(__exec.__device_policy(),
-                                                                     ::std::forward<_Range>(__rng), __comp, __proj);
-}
-
-//------------------------------------------------------------------------
-// parallel_partial_sort
-//-----------------------------------------------------------------------
-
-// TODO: check if it makes sense to move these wrappers out of backend to a common place
-template <typename _ExecutionPolicy, typename _Iterator, typename _Compare,
-          oneapi::dpl::__internal::__enable_if_fpga_execution_policy<_ExecutionPolicy, int> = 0>
-auto
-__parallel_partial_sort(_ExecutionPolicy&& __exec, _Iterator __first, _Iterator __mid, _Iterator __last,
-                        _Compare __comp)
-{
-    // workaround until we implement more performant version for patterns
-    return oneapi::dpl::__par_backend_hetero::__parallel_partial_sort(__exec.__device_policy(), __first, __mid, __last,
-                                                                      __comp);
-}
-
 //------------------------------------------------------------------------
 // parallel_histogram
 //-----------------------------------------------------------------------
 
 // TODO: check if it makes sense to move these wrappers out of backend to a common place
-template <typename _ExecutionPolicy, typename _Event, typename _Range1, typename _Range2, typename _BinHashMgr,
-          oneapi::dpl::__internal::__enable_if_fpga_execution_policy<_ExecutionPolicy, int> = 0>
+template <typename _ExecutionPolicy, typename _Event, typename _Range1, typename _Range2, typename _BinHashMgr>
```

Comment:
@MikeDvorskiy I don't know should we still have this implementation?

Without this static_assert
```C++
    static_assert(sizeof(oneapi::dpl::__internal::__value_t<_Range2>) <= sizeof(::std::uint32_t),
                  "histogram is not supported on FPGA devices with output types greater than 32 bits");
```
I would have deleted it already

Reply:
Actually, this is compile time diagnostic that histogram algorithm doesn't work with fpga policy..
"histogram is not supported on FPGA devices with output types greater than 32 bits"

Why are you going to remove that diagnostic?

Reply:
I think there is no mandatory diagnostic because it's simple recall of `__parallel_histogram` with `__device_backend_tag` which are really implemented for `__device_backend_tag` only.

Reply:
Are you sure that logic (hope you mean run-time) was really changes?
Could you please explain?

Reply:
I mean you should keep "static assert" logic in case if fpga policy and "output types greater than 32 bits" - histogram is not supported.

Reply:
Discussed offline